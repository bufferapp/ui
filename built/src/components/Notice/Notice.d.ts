import React from 'react';
import PropTypes from 'prop-types';
declare function Notice({ children, dismiss, type, className, disableAnimation }: {
    children: any;
    dismiss: any;
    type: any;
    className: any;
    disableAnimation: any;
}): React.JSX.Element;
declare namespace Notice {
    var propTypes: {
        children: PropTypes.Validator<string | number | boolean | PropTypes.ReactElementLike | PropTypes.ReactNodeArray>;
        dismiss: PropTypes.Requireable<(...args: any[]) => any>;
        disableAnimation: PropTypes.Requireable<boolean>;
        type: PropTypes.Validator<string>;
    };
    var defaultProps: {
        dismiss: null;
        disableAnimation: boolean;
    };
}
export default Notice;
//# sourceMappingURL=Notice.d.ts.map