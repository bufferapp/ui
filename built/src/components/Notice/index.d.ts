export { default } from './Notice';
//# sourceMappingURL=index.d.ts.map