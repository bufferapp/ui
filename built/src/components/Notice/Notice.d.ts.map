{"version":3,"file":"Notice.d.ts","sourceRoot":"","sources":["../../../../src/components/Notice/Notice.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAO,SAAS,MAAM,YAAY,CAAA;AAuGlC,iBAAS,MAAM,CAAC,EAAE,QAAQ,EAAE,OAAO,EAAE,IAAI,EAAE,SAAS,EAAE,gBAAgB,EAAE;;;;;;CAAA,qBAgCvE;kBAhCQ,MAAM;;;;;;;;;;;;AAgDf,eAAe,MAAM,CAAA"}