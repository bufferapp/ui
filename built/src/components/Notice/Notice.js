import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { borderRadius } from '../style/borders';
import { fontSize } from '../style/fonts';
import Warning from '../Icon/Icons/Warning';
import Cross from '../Icon/Icons/Cross';
import Buffer from '../Icon/Icons/Buffer';
import { grayDark, grayLighter, grayDarker, redLightest, redDark, redDarker, purpleDark, purpleLightest, purpleDarker, yellowLightest, yellowDark, yellowDarker, } from '../style/colors';
import { useAnimation } from '../AnimationWrapper';
import { stageInRight, fadeOut } from '../style/animations';
const colorMap = {
    alert: {
        border: redDark,
        background: redLightest,
        color: redDarker,
    },
    warning: {
        border: yellowDark,
        background: yellowLightest,
        color: yellowDarker,
    },
    note: {
        border: grayDark,
        background: grayLighter,
        color: grayDarker,
    },
    tip: {
        border: purpleDark,
        background: purpleLightest,
        color: purpleDarker,
    },
};
const NoticeWrapper = styled.div `
  border: ${(props) => `1px solid ${colorMap[props.type].border}`};
  color: ${(props) => colorMap[props.type].color};
  background: ${(props) => colorMap[props.type].background};
  border-radius: ${borderRadius};
  font-size: ${fontSize};
  padding: 8px ${({ dismiss }) => (dismiss ? '28px' : '8px')} 8px 8px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  width: 100%;
`;
const WarningIcon = styled(Warning) `
  display: block;
  margin-right: 12px;
  flex: 1 0 auto;
  max-width: 16px;
`;
const BufferIcon = styled(Buffer) `
  display: block;
  margin-right: 12px;
  flex: 1 0 auto;
  max-width: 16px;
`;
const CloseButton = styled.button `
  color: ${(props) => colorMap[props.type].color};
  border: 0;
  background: 0;
  padding: 0;
  margin: 0;
  opacity: 0.8;
  height: 16px;
  position: absolute;
  right: 16px;
  &:hover {
    color: ${(props) => colorMap[props.type].color};
    opacity: 1;
    cursor: pointer;
  }
`;
function Notice({ children, dismiss, type, className, disableAnimation }) {
    const { AnimationWrapper, dismiss: dismissAnimationWrapper, animationProps, } = useAnimation({
        justify: 'flex-end',
        stageInAnimation: disableAnimation ? undefined : stageInRight,
        stageOutAnimation: disableAnimation ? undefined : fadeOut,
        onDismiss: dismiss,
    });
    const noticeContent = (React.createElement(NoticeWrapper, { type: type, dismiss: dismiss, className: className },
        type === 'warning' && React.createElement(WarningIcon, null),
        type === 'alert' && React.createElement(WarningIcon, null),
        type === 'tip' && React.createElement(BufferIcon, null),
        children,
        dismiss && (React.createElement(CloseButton, { type: type, onClick: () => dismissAnimationWrapper() },
            React.createElement(Cross, null)))));
    return (React.createElement(AnimationWrapper, Object.assign({}, animationProps), noticeContent));
}
Notice.propTypes = {
    children: PropTypes.node.isRequired,
    dismiss: PropTypes.func,
    disableAnimation: PropTypes.bool,
    type: PropTypes.string.isRequired,
};
Notice.defaultProps = {
    dismiss: null,
    disableAnimation: false,
};
export default Notice;
//# sourceMappingURL=Notice.js.map