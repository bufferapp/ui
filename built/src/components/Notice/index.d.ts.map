{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/Notice/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,UAAU,CAAA"}