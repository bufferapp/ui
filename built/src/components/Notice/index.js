export { default } from './Notice';
//# sourceMappingURL=index.js.map