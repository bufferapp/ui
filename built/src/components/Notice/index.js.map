{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/Notice/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,UAAU,CAAA","sourcesContent":["export { default } from './Notice'\n"]}