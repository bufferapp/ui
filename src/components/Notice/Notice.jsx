import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { borderRadius } from '../style/borders'
import { fontSize } from '../style/fonts'
import Warning from '../Icon/Icons/Warning'
import Cross from '../Icon/Icons/Cross'
import Buffer from '../Icon/Icons/Buffer'
import {
  grayDark,
  grayLighter,
  grayDarker,
  redLightest,
  redDark,
  redDarker,
  purpleDark,
  purpleLightest,
  purpleDarker,
  yellowLightest,
  yellowDark,
  yellowDarker,
} from '../style/colors'
import { useAnimation } from '../AnimationWrapper'
import { stageInRight, fadeOut } from '../style/animations'

const colorMap = {
  alert: {
    border: redDark,
    background: redLightest,
    color: redDarker,
  },
  warning: {
    border: yellowDark,
    background: yellowLightest,
    color: yellowDarker,
  },
  note: {
    border: grayDark,
    background: grayLighter,
    color: grayDarker,
  },
  tip: {
    border: purpleDark,
    background: purpleLightest,
    color: purpleDarker,
  },
}

const NoticeWrapper = styled.div`
  border: ${(props) => `1px solid ${colorMap[props.type].border}`};
  color: ${(props) => colorMap[props.type].color};
  background: ${(props) => colorMap[props.type].background};
  border-radius: ${borderRadius};
  font-size: ${fontSize};
  padding: 8px ${({ dismiss }) => (dismiss ? '28px' : '8px')} 8px 8px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  width: 100%;
`

const WarningIcon = styled(Warning)`
  display: block;
  margin-right: 12px;
  flex: 1 0 auto;
  max-width: 16px;
`

const BufferIcon = styled(Buffer)`
  display: block;
  margin-right: 12px;
  flex: 1 0 auto;
  max-width: 16px;
`

const CloseButton = styled.button`
  color: ${(props) => colorMap[props.type].color};
  border: 0;
  background: 0;
  padding: 0;
  margin: 0;
  opacity: 0.8;
  height: 16px;
  position: absolute;
  right: 16px;
  &:hover {
    color: ${(props) => colorMap[props.type].color};
    opacity: 1;
    cursor: pointer;
  }
`

function Notice({ children, dismiss, type, className, disableAnimation }) {
  // We always need to wrap the Notice with AnimationWrapper because the dismiss function
  // shows the Notice based on the dismissed property. If the animation is disabled the properties
  // that control the animation are removed from animationProps.
  const {
    AnimationWrapper,
    dismiss: dismissAnimationWrapper,
    animationProps,
  } = useAnimation({
    justify: 'flex-end',
    stageInAnimation: disableAnimation ? undefined : stageInRight,
    stageOutAnimation: disableAnimation ? undefined : fadeOut,
    onDismiss: dismiss,
  })

  const noticeContent = (
    <NoticeWrapper type={type} dismiss={dismiss} className={className}>
      {type === 'warning' && <WarningIcon />}
      {type === 'alert' && <WarningIcon />}
      {type === 'tip' && <BufferIcon />}
      {children}
      {dismiss && (
        <CloseButton type={type} onClick={() => dismissAnimationWrapper()}>
          <Cross />
        </CloseButton>
      )}
    </NoticeWrapper>
  )

  return (
    <AnimationWrapper {...animationProps}>{noticeContent}</AnimationWrapper>
  )
}

Notice.propTypes = {
  children: PropTypes.node.isRequired,
  dismiss: PropTypes.func,
  /** doesn't use animation wrapper if true */
  disableAnimation: PropTypes.bool,
  /** can be warning, note, alert, tip */
  type: PropTypes.string.isRequired,
}

Notice.defaultProps = {
  dismiss: null,
  disableAnimation: false,
}

export default Notice
