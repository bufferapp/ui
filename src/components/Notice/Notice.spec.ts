/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Notice from './Notice'

snap(Notice, './Notice.jsx')
