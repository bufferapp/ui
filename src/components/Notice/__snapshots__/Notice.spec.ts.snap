// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Notice Matches snapshot when boolean prop "disableAnimation" is set to: "false" 1`] = `
<AnimationWrapper
  align="center"
  dismissing={false}
  duration={300}
  easing="cubic-bezier(0.25, 1, 0.5, 1)"
  justify="flex-end"
  onDismiss={[MockFunction]}
  stageInAnimation={
    e {
      "id": "sc-keyframes-jALQVJ",
      "inject": [Function],
      "name": "jALQVJ",
      "rules": "
  0% {
    transform: translateX(200px);
    opacity: 0;
  }

  100% {
    transform: translateX(0);
    opacity: 1;
  }
",
      "toString": [Function],
    }
  }
  stageOutAnimation={
    e {
      "id": "sc-keyframes-bjhQjg",
      "inject": [Function],
      "name": "bjhQjg",
      "rules": "
  0% {
    transform: scale(1) translateY(0);
    opacity: 1;
  }

  100% {
    transform: scale(1.05) translateY(-10px);
    opacity: 0;
  }
",
      "toString": [Function],
    }
  }
>
  <styled.div
    dismiss={[MockFunction]}
    type="jest-auto-snapshots String Fixture"
  >
    <NodeFixture />
    <styled.button
      onClick={[Function]}
      type="jest-auto-snapshots String Fixture"
    >
      <CrossIcon />
    </styled.button>
  </styled.div>
</AnimationWrapper>
`;

exports[`jest-auto-snapshots > Notice Matches snapshot when passed all props 1`] = `
<AnimationWrapper
  align="center"
  dismissing={false}
  duration={300}
  easing="cubic-bezier(0.25, 1, 0.5, 1)"
  justify="flex-end"
  onDismiss={[MockFunction]}
>
  <styled.div
    dismiss={[MockFunction]}
    type="jest-auto-snapshots String Fixture"
  >
    <NodeFixture />
    <styled.button
      onClick={[Function]}
      type="jest-auto-snapshots String Fixture"
    >
      <CrossIcon />
    </styled.button>
  </styled.div>
</AnimationWrapper>
`;

exports[`jest-auto-snapshots > Notice Matches snapshot when passed only required props 1`] = `
<AnimationWrapper
  align="center"
  dismissing={false}
  duration={300}
  easing="cubic-bezier(0.25, 1, 0.5, 1)"
  justify="flex-end"
  onDismiss={null}
  stageInAnimation={
    e {
      "id": "sc-keyframes-jALQVJ",
      "inject": [Function],
      "name": "jALQVJ",
      "rules": "
  0% {
    transform: translateX(200px);
    opacity: 0;
  }

  100% {
    transform: translateX(0);
    opacity: 1;
  }
",
      "toString": [Function],
    }
  }
  stageOutAnimation={
    e {
      "id": "sc-keyframes-bjhQjg",
      "inject": [Function],
      "name": "bjhQjg",
      "rules": "
  0% {
    transform: scale(1) translateY(0);
    opacity: 1;
  }

  100% {
    transform: scale(1.05) translateY(-10px);
    opacity: 0;
  }
",
      "toString": [Function],
    }
  }
>
  <styled.div
    dismiss={null}
    type="jest-auto-snapshots String Fixture"
  >
    <NodeFixture />
  </styled.div>
</AnimationWrapper>
`;
